### ISSUE TITLE
[folly] build failure

### ISSUE THREAD

**Issue by LLlKuIIeP** [microsoft/vcpkg#9156]
**Host Environment**
- OS: Windows 10 x64
- Compiler:  Microsoft (R) C/C++ версии 19.00.24234.1 для x86

**To Reproduce**
`.\vcpkg install folly`

**Failure logs** 
-PATH\vcpkg>.\vcpkg install folly
The following packages will be built and installed:
    folly[core,zlib]:x86-windows
Starting package 1/1: folly:x86-windows
Building package folly[core,zlib]:x86-windows...
CMake Error at ports/folly/portfile.cmake:2 (message):
  Folly only supports the x64 architecture.
Call Stack (most recent call first):
  scripts/ports.cmake:94 (include)

Error: Building package folly:x86-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: folly:x86-windows
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.

**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

**Comment by timautry** [comment]
Looks like folly only supports 64 bit architecture, but the default triplet for vcpkg is 32 bit architecture (aka x86-windows triplet).

From the vcpkg root folder, try  **_.\vcpkg.exe install folly[core,zlib]:x64-windows_**

**Comment by LLlKuIIeP** [comment]
> Looks like folly only supports 64 bit architecture, but the default triplet for vcpkg is 32 bit architecture (aka x86-windows triplet).
> 
> From the vcpkg root folder, try **_.\vcpkg.exe install folly[core,zlib]:x64-windows_**

Thanks, it helped me for **folly**.  And for **fizz** what to do?

C:\Repositories\vcpkg>.\vcpkg install fizz[core]:x64-windows
The following packages will be built and installed:
    fizz[core]:x64-windows
Starting package 1/1: fizz:x64-windows
Building package fizz[core]:x64-windows...
-- Note: fizz only supports static library linkage. Building static library.
-- Using cached C:/Repositories/vcpkg/downloads/facebookincubator-fizz-3e933cff04c27c97d0fb0b899ed0c883b4e02e3b.tar.gz
-- Using source at C:/Repositories/vcpkg/buildtrees/fizz/src/83b4e02e3b-e4b25f2833
-- Configuring x64-windows
-- Building x64-windows-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:136 (message):
    Command failed: "C:/Program Files/CMake/bin/cmake.exe" --build . --config Debug --target install -- -v
    Working Directory: C:/Repositories/vcpkg/buildtrees/fizz/x64-windows-dbg
    See logs for more information:
      C:\Repositories\vcpkg\buildtrees\fizz\install-x64-windows-dbg-out.log

Call Stack (most recent call first):
  scripts/cmake/vcpkg_build_cmake.cmake:91 (vcpkg_execute_build_process)
  scripts/cmake/vcpkg_install_cmake.cmake:24 (vcpkg_build_cmake)
  ports/fizz/portfile.cmake:30 (vcpkg_install_cmake)
  scripts/ports.cmake:94 (include)


Error: Building package fizz:x64-windows failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: fizz:x64-windows
  Vcpkg version: 2019.09.12-nohash

Additionally, attach any relevant sections from the log files above.

**Comment by timautry** [comment]
Please attach the file (or post the contents of, if it is small) **C:\Repositories\vcpkg\buildtrees\fizz\install-x64-windows-dbg-out.log**. 
This may help with one more knowledgeable than I in assisting your issue :)

**Comment by LLlKuIIeP** [comment]
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/3934898/install-x64-windows-dbg-out.log)

**Comment by LLlKuIIeP** [comment]
[install-x64-windows-dbg-out.log](https://github.com/microsoft/vcpkg/files/3934905/install-x64-windows-dbg-out.log)

**Comment by JackBoosY** [comment]
@LLlKuIIeP Seems like a Visual Studio bug, please update Visual Studio and try again.

Thanks.

**Comment by LLlKuIIeP** [comment]
Thanks, but are the compilers for Studios different or the same? Because even before the installation began, I updated Visual Studio 17th and 19th using vs_community__1715138479.1548661295.

**Comment by JackBoosY** [comment]
@LLlKuIIeP My Visual Studio version is 15.9.17 and I cannot repro it.
Can you provide your Visual Studio version? You can generate it in Visual Studio menu-Help-About Microsoft Visual Studio-Copy Info.
![image](https://user-images.githubusercontent.com/47264268/70421292-81088780-1aa4-11ea-92f2-28745dbc6f28.png)

Thanks.

**Comment by LLlKuIIeP** [comment]
Microsoft Visual Studio Community 2017 
Версия 15.9.17
VisualStudio.15.Release/15.9.17+28307.905
Microsoft .NET Framework
Версия 4.8.03752

Установленная версия: Community

Visual C++ 2017   00369-60000-00001-AA703
Microsoft Visual C++ 2017

ASP.NET and Web Tools 2017   15.9.04012.0
ASP.NET and Web Tools 2017

Intel® VTune™ Amplifier 2019 Update 8   604197
Intel® VTune™ Amplifier 2019 Update 8, (build 604197),  Copyright © 2009-2019 Intel Corporation. All rights reserved.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

MLGen Package Extension   1.0
MLGen Package Visual Studio Extension Detailed Info

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Qt Visual Studio Tools   2.4.2
The Qt Visual Studio Tools allow developers to use the standard development environment without having to worry about any Qt-related build steps or tools.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

WiX Toolset Visual Studio Extension   1.0.0.4
WiX Toolset Visual Studio Extension version 1.0.0.4
Copyright (c) .NET Foundation and contributors. All rights reserved.

Адаптер теста для Google Test   1.0
Включает средства тестирования Visual Studio и модульные тесты для Google Test.  Условия использования и уведомления сторонних производителей см. в каталоге установки расширения.

Диспетчер пакетов NuGet   4.6.0
Диспетчер пакетов NuGet в Visual Studio. Дополнительную информацию о NuGet см. на веб-сайте http://docs.nuget.org/.

Инструменты Visual Studio для CMake   1.0
Инструменты Visual Studio для CMake

Мастера Microsoft Visual C++   1.0
Мастера Microsoft Visual C++

Пакет VC Microsoft Visual Studio   1.0
Пакет VC Microsoft Visual Studio

Пакет размещения адаптеров отладки Visual Studio Code   1.0
Уровень взаимодействия для размещения адаптеров отладки Visual Studio Code в Visual Studio

Средства C#   2.10.0-beta2-63501-03+b9fb1610c87cccc8ceb74a770dba261a58e39c4a
Компоненты C#, используемые в среде IDE. В зависимости от типа и параметров проекта может использоваться другая версия компилятора.

Средства Visual Basic   2.10.0-beta2-63501-03+b9fb1610c87cccc8ceb74a770dba261a58e39c4a
Компоненты Visual Basic, используемые в среде IDE. В зависимости от типа и параметров проекта может использоваться другая версия компилятора.

Языковая служба JavaScript   2.0
Языковая служба JavaScript

**Comment by JackBoosY** [comment]
We need the version info of Visual Studio 2019, please provide Visual Studio 2019 version info.

Thanks.

**Comment by LLlKuIIeP** [comment]
Microsoft Visual Studio Community 2019
Версия 16.3.10
VisualStudio.16.Release/16.3.10+29519.87
Microsoft .NET Framework
Версия 4.8.03752

Установленная версия: Community

Visual C++ 2019   00435-60000-00000-AA760
Microsoft Visual C++ 2019

ASP.NET and Web Tools 2019   16.3.286.43615
ASP.NET and Web Tools 2019

Intel® VTune™ Amplifier 2019 Update 8   604197
Intel® VTune™ Amplifier 2019 Update 8, (build 604197),  Copyright © 2009-2019 Intel Corporation. All rights reserved.

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Qt Visual Studio Tools   2.3.4
The Qt Visual Studio Tools allow developers to use the standard development environment without having to worry about any Qt-related build steps or tools.

Test Adapter for Boost.Test   1.0
Enables Visual Studio's testing tools with unit tests written for Boost.Test.  The use terms and Third Party Notices are available in the extension installation directory.

Visual C++ для кроссплатформенной разработки мобильных приложений   16.0.29230.54
Visual C++ для кроссплатформенной разработки мобильных приложений

Адаптер теста для Google Test   1.0
Включает средства тестирования Visual Studio и модульные тесты для Google Test.  Условия использования и уведомления сторонних производителей см. в каталоге установки расширения.

Диспетчер пакетов NuGet   5.3.1
Диспетчер пакетов NuGet в Visual Studio. Дополнительные сведения о NuGet см. на веб-сайте https://docs.nuget.org/

Инструменты C#   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
Компоненты C#, используемые в интегрированной среде разработки. В зависимости от типа и настроек проекта могут использоваться различные версии компилятора.

Инструменты Visual Basic   3.3.1-beta3-19461-02+2fd12c210e22f7d6245805c60340f6a34af6875b
Компоненты Visual Basic, используемые в интегрированной среде разработки. В зависимости от типа и настроек проекта могут использоваться различные версии компилятора.

Инструменты Visual Studio для CMake   1.0
Инструменты Visual Studio для CMake

Инструменты Visual Studio для CMake   1.0
Инструменты Visual Studio для CMake

Мастера Microsoft Visual C++   1.0
Мастера Microsoft Visual C++

Пакет VC Microsoft Visual Studio   1.0
Пакет VC Microsoft Visual Studio

Пакет размещения адаптеров отладки Visual Studio Code   1.0
Уровень взаимодействия для размещения адаптеров отладки Visual Studio Code в Visual Studio

Разработка на Visual C++ для Linux   1.0.9.29424
Разработка на Visual C++ для Linux

Расширение IntelliCode   1.0
Подробные сведения о расширении IntelliCode для Visual Studio

**Comment by LLlKuIIeP** [comment]
My colleague installed fizz and wangle libraries, but he has Microsoft Visual Studio 12.0.

Thanks.

**Comment by JackBoosY** [comment]
Block in this problem:
```
F:\vcpkg\installed\x64-windows\include\boost\variant\detail\forced_return.hpp(43) : error C4716: 'boost::detail::variant::forced_return<std::unique_ptr<folly::futures::detail::DeferredExecutor,folly::futures::detail::UniqueDeleter> const *>': must return a value
F:\vcpkg\installed\x64-windows\include\boost\variant\detail\forced_return.hpp(43) : error C4716: 'boost::detail::variant::forced_return<std::unique_ptr<folly::futures::detail::DeferredExecutor,folly::futures::detail::UniqueDeleter> *>': must return a value
F:\vcpkg\installed\x64-windows\include\boost\variant\detail\forced_return.hpp(43) : error C4716: 'boost::detail::variant::forced_return<folly::Executor::KeepAlive<folly::Executor> const *>': must return a value
F:\vcpkg\installed\x64-windows\include\boost\variant\detail\forced_return.hpp(43) : error C4716: 'boost::detail::variant::forced_return<folly::Executor::KeepAlive<folly::Executor> *>': must return a value
```
Solved using PR [#4](https://github.com/boostorg/variant/pull/4).

But I still can't reproduce this issue, my Visual Studio 2019 version is 16.4.0, this error should be fixed in the latest version. So please update Visual Studio 2019 to the latest version and try again.

Thanks.

**Comment by LLlKuIIeP** [comment]
Thanks you very match.
C:\Repositories\vcpkg> .\vcpkg.exe install fizz[core]:x64-windows
...
Installing package fizz[core]:x64-windows... done

And

C:\Repositories\vcpkg> .\vcpkg.exe install wangle[core]:x64-windows
Installing package wangle[core]:x64-windows... done
